`zmq.error.ZMQError: Invalid argument` when mocking subscriber

Python-3.8.1 (default, Feb  2 2020, 08:37:37)
libzmq-4.3.2
pyzmq-18.1.1

I have problem with mocking the subscriber - it works fine on Python 3.7 and 3.6, the problem appeared after updating to Python 3.8. Not entirely sure if it's a problem with mock or pyzmq - I don't have problems with other mocks, but I also don't have problem with zmq when running the app normally.

Code causing error:
```python
class Receiver:
    def __init__(self, relay):
        self.relay = relay
        context = zmq.Context.instance()
        self.subscriber = context.socket(zmq.SUB)
        self.subscriber.setsockopt(zmq.SUBSCRIBE, b"")
        logger.debug(f'Receiver created with relay {relay}')
        self.connect()

    def connect(self):
        logger.debug(f'Connecting to {self.relay}...')
        self.subscriber.connect(self.relay)

@pytest.fixture()
def mock_receiver(relay)
    receiver = Receiver(relay=relay))
    mock_subscriber = Mock(spec=receiver.subscriber)
```
When running tests, on this line `mock_subscriber = Mock(spec=receiver.subscriber)` I get the following error:

```
conftest.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.8/unittest/mock.py:1057: in __init__
    _safe_super(CallableMixin, self).__init__(
/usr/local/lib/python3.8/unittest/mock.py:442: in __init__
    self._mock_add_spec(spec, spec_set, _spec_as_instance, _eat_self)
/usr/local/lib/python3.8/unittest/mock.py:497: in _mock_add_spec
    if asyncio.iscoroutinefunction(getattr(spec, attr, None)):
/usr/local/lib/python3.8/site-packages/zmq/sugar/attrsettr.py:48: in __getattr__
    return self._get_attr_opt(upper_key, opt)
/usr/local/lib/python3.8/site-packages/zmq/sugar/attrsettr.py:52: in _get_attr_opt
    return self.get(opt)
zmq/backend/cython/socket.pyx:476: in zmq.backend.cython.socket.Socket.get
    ???
zmq/backend/cython/socket.pyx:262: in zmq.backend.cython.socket._getsockopt
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   zmq.error.ZMQError: Invalid argument

zmq/backend/cython/checkrc.pxd:25: ZMQError
```

As I said I'm not enitrely sure if the problem is on pyzmq side, sorry if I'm wasting your time, I'm a newbie with pyzmq. In case that's useful, here are the libraries I use for testing (all up to date).
mock==4.0.1
pytest==5.3.5
pytest-mock==2.0.0

--- a/zmq/sugar/attrsettr.py
+++ b/zmq/sugar/attrsettr.py
@@ -4,7 +4,7 @@
 # Copyright (C) PyZMQ Developers
 # Distributed under the terms of the Modified BSD License.
 
-
+import errno
 from . import constants
 
 class AttributeSetter(object):
@@ -45,7 +45,18 @@ class AttributeSetter(object):
                 self.__class__.__name__, upper_key)
             )
         else:
-            return self._get_attr_opt(upper_key, opt)
+            from zmq import ZMQError
+            try:
+                return self._get_attr_opt(upper_key, opt)
+            except ZMQError as e:
+                # EINVAL will be raised on access for write-only attributes.
+                # Turn that into an AttributeError
+                # necessary for mocking
+                if e.errno == errno.EINVAL:
+                    raise AttributeError("{} attribute is write-only".format(key))
+                else:
+                    raise
+
 
     def _get_attr_opt(self, name, opt):
         """override if getattr should do something other than call self.get"""

--- a/zmq/tests/test_context.py
+++ b/zmq/tests/test_context.py
@@ -11,6 +11,10 @@ try:
     from queue import Queue
 except ImportError:
     from Queue import Queue
+try:
+    from unittest import mock
+except ImportError:
+    mock = None
 
 from pytest import mark
 
@@ -51,6 +55,11 @@ class TestContext(BaseZMQTestCase):
         if zmq.zmq_version_info() > (3,):
             self.assertTrue('IO_THREADS' in dir(ctx))
         ctx.term()
+
+    @mark.skipif(mock is None, reason="requires unittest.mock")
+    def test_mockable(self):
+        m = mock.Mock(spec=self.context)
+
 
     def test_term(self):
         c = self.Context()

--- a/zmq/tests/test_socket.py
+++ b/zmq/tests/test_socket.py
@@ -11,6 +11,10 @@ import time
 import warnings
 import socket
 import sys
+try:
+    from unittest import mock
+except ImportError:
+    mock = None
 
 import pytest
 from pytest import mark
@@ -54,7 +58,7 @@ class TestSocket(BaseZMQTestCase):
                 self.assertEqual(b.closed, True)
             self.assertEqual(a.closed, True)
         self.assertEqual(ctx.closed, True)
-    
+
     def test_dir(self):
         ctx = self.Context()
         s = ctx.socket(zmq.PUB)
@@ -64,6 +68,12 @@ class TestSocket(BaseZMQTestCase):
         self.assertTrue('FD' in dir(s))
         s.close()
         ctx.term()
+
+    @mark.skipif(mock is None, reason="requires unittest.mock")
+    def test_mockable(self):
+        s = self.socket(zmq.SUB)
+        m = mock.Mock(spec=s)
+        s.close()
 
     def test_bind_unicode(self):
         s = self.socket(zmq.PUB)

